docs/tools/testing/testing_greentea.md

```diff
@@ -178,6 +322,193 @@ This detects your attached target and drives the test. If you need to rerun the
+
+#### Executing all tests
+
+The default action of Greentea using `mbed test` is to execute all tests that are found. You can also add `-v` to make the output more verbose.
+
+#### Limiting tests
+
+You can select test cases by name using the `-n` argument. This command executes all tests named `tests-mbedmicro-rtos-mbed-mail` from all builds in the test specification:
+
+```
+$ mbed test -n tests-mbedmicro-rtos-mbed-mail
+```
+
+When using the `-n` argument, you can use the `*` character as a wildcard. This command executes all tests that start with `tests-` and have `-rtos-` in them.
+
+```
+$ mbed test -n tests-*-rtos-*
+```
+
+You can use a comma (`,`) to separate test names (argument `-n`) and build names (argument `-t`). This command executes the tests `tests-mbedmicro-rtos-mbed-mail` and `tests-mbed_drivers-c_strings` for the `K64F-ARM` and `K64F-GCC_ARM` builds in the test specification:
+
+```
+$ mbed test -n tests-mbedmicro-rtos-mbed-mail,tests-mbed_drivers-c_strings -t K64F-ARM,K64F-GCC_ARM
+```
+
+#### Selecting platforms
+
+You can limit which boards Greentea uses for testing by using the `--use-tids` argument.
+
+```
+$ mbed test --use-tids 02400203C3423E603EBEC3D8,024002031E031E6AE3FFE3D2 --run
+```
+
+Where `02400203C3423E603EBEC3D8` and `024002031E031E6AE3FFE3D` are the target IDs of platforms attached to your system.
+
+You can view target IDs using [mbed-ls](https://github.com/ARMmbed/mbed-os-tools/tree/master/packages/mbed-ls), which is installed as part of Mbed CLI.
+
+```
+$ mbedls
++--------------+---------------------+------------+------------+-------------------------+
+|platform_name |platform_name_unique |mount_point |serial_port |target_id                |
++--------------+---------------------+------------+------------+-------------------------+
+|K64F          |K64F[0]              |E:          |COM160      |024002031E031E6AE3FFE3D2 |
+|K64F          |K64F[1]              |F:          |COM162      |02400203C3423E603EBEC3D8 |
+|LPC1768       |LPC1768[0]           |G:          |COM5        |1010ac87cfc4f23c4c57438d |
++--------------+---------------------+------------+------------+-------------------------+
+```
+
+In this case, you won't test one target, the LPC1768.
+
+#### Creating reports
+
+Greentea supports a number of report formats.
+
+##### HTML
+
+This creates an interactive HTML page with test results and logs.
+
+```
+mbed test --report-html html_report.html --run
+```
+
+##### JUnit
+
+This creates an XML JUnit report, which you can use with popular Continuous Integration software, such as [Jenkins](https://jenkins.io/index.html).
+
+```
+mbed test --report-junit junit_report.xml --run
+```
+
+##### JSON
+
+This creates a general JSON report.
+
+```
+mbed test --report-json json_report.json --run
+```
+
+##### Plain text
+
+This creates a human-friendly text summary of the test run.
+
+```
+mbed test --report-text text_report.text --run
```

Comment:
Thinking about this more, none of the report arguments are currently part of the help options in Mbed CLI. This only works because Mbed CLI passes along arguments to the underlying scripts when it doesn't recognize it. How do you feel about removing this reporting section altogether?